main.go

```diff
@@ -246,11 +332,17 @@ func (m *Miner) hashratePrinter() {
 }
 
 // resultLoop takes in the result and passes to the proper channels for receiving.
-func (m *Miner) resultLoop() error {
+func (m *Miner) resultLoop(config util.Config) {
 	for {
 		select {
 		case header := <-m.resultCh:
 			order, err := m.GetDifficultyOrder(header)
+			if !config.Proxy {
+				go m.sendMinedHeaderNodes(order, header)
```

Comment:
Addressed a verbal comment that indicated preference for calculating and using order in the same place.
Simplified sendMinedHeaderNodes to only send to one destination, and the for loop calls that function multiple times as needed.